lib/matplotlib/widgets.py

```diff
@@ -583,6 +596,18 @@ def on_changed(self, func):
         """
         return self._observers.connect('changed', lambda val: func(val))
 
+    def set_limits(self, vmin=None, vmax=None):
+        """Update the limits of the slider."""
+        super().set_limits(vmin=vmin, vmax=vmax)
+        self.val = self._value_in_bounds(self.val)
+        # if we reset the slider after updating the limits then we should have
+        # the proper valinit value
+        self.valinit = self._value_in_bounds(self.valinit)
+        if self.orientation == 'vertical':
+            self.hline.set_ydata(self.valinit)
+        else:
+            self.vline.set_xdata(self.valinit)
+
 
 class RangeSlider(SliderBase):
```

Comment:
Does `RangeSlider` also need some custom `set_limits`, or is the base class implementation enough?